### PR TITLE
[weather.gismeteo@krypton] 0.4.1

### PR DESCRIPTION
**Pull request by vlmaksime**
### Add-on details:

- **General**
  - Add-on name: Gismeteo
  - Add-on ID: weather.gismeteo
  - Version number: 0.4.1
  - Kodi/repository version: krypton

- **Code location**
  - URL: https://github.com/vlmaksime/weather.gismeteo
  
The weather is provided solely for personal non-commercial use

### Description of changes:

v0.4.1 (2020-05-03)
- Changed data caching mechanism
- Fixed identified bugs
- Added Spanish translation (CHIKIRY WIZARD GROUP)

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

### CODE REVIEW

File: weather.gismeteo/addon.xml [hunk 1]
```diff
@@ -1,27 +1,30 @@
...
+		<summary lang="ru_RU">Прогноз погоды от ведущего погодного сервиса Gismeteo</summary>
+		<summary lang="es_ES">Pronóstico del tiempo Gismeteo. Clima en tiempo real y pronóstico detallado en todo el mundo</summary>
+		<description lang="en_GB">The weather is provided solely for personal non-commercial use</description>
+		<description lang="ru_RU">Погода предоставлена исключительно для личного некоммерческого использования</description>
+		<description lang="es_ES">El clima se proporciona únicamente para su uso personal y no comercial</description>
+		<disclaimer lang="en_GB">The weather forecast has been prepared by MapMakers Group Ltd</disclaimer>
+		<disclaimer lang="ru_RU">Прогноз погоды подготовлен НПЦ "Мэп Мейкер"</disclaimer>
+		<disclaimer lang="es_ES">El pronóstico del tiempo ha sido preparado por MapMakers Group Ltd</disclaimer>
+		<platform>all</platform>
+		<license>GNU GENERAL PUBLIC LICENSE. Version 3, 29 June 2007</license>
```
Comment (basrieter) [inline weather.gismeteo/addon.xml:20]: Please use the corresponding SPDX from https://spdx.org/licenses/

File: weather.gismeteo/resources/libs/gismeteo.py [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+class GismeteoClient(object):
+
+    _base_url = 'https://services.gismeteo.ru/inform-service/inf_chrome'
+
+    def __init__(self, lang='en'):
+
+        self._lang = lang
+
+        self._client = requests.Session()
+        if PY26:
```
Comment (basrieter) [inline weather.gismeteo/resources/libs/gismeteo.py:34]: Why only PY26? Would it not be sufficient to check PY2 or PY3?

File: weather.gismeteo/resources/libs/simpleweather.py [hunk 3]
```diff
@@ -0,0 +1,725 @@
...
+        xbmcgui.Dialog().notification(_heading, _message, icon, time, sound)
+
+    def dialog_ok(self, line1, line2="", line3=""):
+
+        if self.kodi_major_version() >= '19':
+            xbmcgui.Dialog().ok(self.name, self._join_strings(line1, line2, line3))
+        else:
+            xbmcgui.Dialog().ok(self.name, line1, line2, line3)
+
+    def dialog_progress_create(self, heading, line1="", line2="", line3=""):
```
Comment (basrieter) [inline weather.gismeteo/resources/libs/simpleweather.py:442]: As a small reminder: the line1, line2, line3 will be removed in Matrix: https://forum.kodi.tv/showthread.php?tid=344263

File: weather.gismeteo/resources/libs/gismeteo.py (hunk 2 above)
Comment (vlmaksime) [reply weather.gismeteo/resources/libs/gismeteo.py:34]: I had an error with SSL certificate on my old android box with Kodi 16 and python 2.6, so I add check for this version. But this code actual for all python versions, so I will remove this check.

File: weather.gismeteo/resources/libs/simpleweather.py (hunk 3 above)
Comment (vlmaksime) [reply weather.gismeteo/resources/libs/simpleweather.py:442]: For Matrix and newer, I use `self._join_strings()` to join these lines into one

File: weather.gismeteo/resources/libs/gismeteo.py [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+class GismeteoClient(object):
+
+    _base_url = 'https://services.gismeteo.ru/inform-service/inf_chrome'
+
+    def __init__(self, lang='en'):
+
+        self._lang = lang
+
+        self._client = requests.Session()
+        self._client.verify = False
```
Comment (basrieter) [inline weather.gismeteo/resources/libs/gismeteo.py:33]: This is not the way to solve SSL issues. You are basically disabling the security of the SSL mechanism.

Comment (vlmaksime) [reply weather.gismeteo/resources/libs/gismeteo.py:33]: There was this error on some platforms :

> hostname 'services.gismeteo.ru' doesn't match either of '*.gismeteo.net', 'gismeteo.net'

Looks like changing URL from services.gismeteo.ru to services.gismeteo.net solves it. I didn’t think about such decision at that moment

**Pull request merged**